editoast/src/core/simulation.rs

```diff
@@ -66,25 +66,41 @@ pub struct PhysicsRollingStock {
     pub raise_pantograph_time: Option<u64>,
 }
 
-impl From<RollingStockModel> for PhysicsRollingStock {
-    fn from(value: RollingStockModel) -> Self {
+#[derive(Debug, Default)]
+pub struct SimulationParameters {
+    pub total_mass: Option<f64>,
+    pub total_length: Option<f64>,
+}
+
+impl PhysicsRollingStock {
+    /// Create a new consist with simulation parameters
+    pub fn new(traction_engine: RollingStockModel, params: SimulationParameters) -> Self {
+        let traction_engine_length = traction_engine.length * 1000.0;
+        let length = params
+            .total_length
+            .unwrap_or(traction_engine_length)
+            .round() as u64;
```

Comment:
Why the *1000 only in a single case? Assuming it's correct, I'm not sure having naked floats with two different length units is a good idea...

Reply:
https://github.com/OpenRailAssociation/osrd/pull/9106#discussion_r1784504055